### PR TITLE
Feature include .git folder in s2i build

### PR DESCRIPTION
**Pull request by eformat**
### Description
Quite often developers use the local git info during the build e.g. in Java land we use the git-commit-id-maven-plugin.

During s2i build we need to pass two flags to include the .git directory.

This change adds the '--copy' and ''--exclude=""' parameters to the s2i-build action. s2i by default will attempt to "git clone" your repo, even if it's a local repo. The relevant discussions and changes for s2i here were discussed here:

-- s2i build --copy (not exposed by action, so it always clones head)
https://github.com/openshift/source-to-image/issues/674#issuecomment-270557199
-- s2i build tar excludes .git when copying to build container
https://github.com/openshift/source-to-image/pull/460

### Related Issue(s)
N/A

### Checklist

- [X] This PR includes a documentation change
- [ ] This PR does not need a documentation change
---
- [ ] This PR includes test changes
- [ ] This PR's changes are already tested
---
- [ ] This change is not user-facing
- [X] This change is a patch change
- [ ] This change is a minor change
- [ ] This change is a major (breaking) change

### Changes made
- added optional argument `include_git` which can be 'false' (anything else is considered 'true')

```yaml
      # Setup S2i and Build container image from source
      - name: Setup and Build
        id: build_image
        uses: redhat-actions/s2i-build@main
        with:
          path_context: '.'
          # Builder image for a java project
          builder_image: 'registry.access.redhat.com/ubi8/openjdk-17:latest'
          image: ${{ env.IMAGE_NAME }}
          tags: ${{ env.TAGS }}
          log_level: 5
          include_git: true
```

I'm not sure how the dist/ works for these actions repo's, i ran

```bash
npm run clean && npm run compile && npm run bundle
```
and tested the changes. Please let me know if i need to change anything in the PR to suit how you test & deploy them.

### CODE REVIEW

**Comment by divyansh42** [comment]
@eformat Thanks for the PR.
You have to do
`npm run lint` --> as CI checks include ESLint too.
`npx action-io-generator --actionYml ./action.yml --outFile ./src/generated/inputs-outputs.ts` --> to generate inputs and ouptuts file using action.yml. Also make sure you do not edit `./src/generated/inputs-outputs.ts` manually.
`npm run bundle` to update `dist/`

File: src/s2iExec.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export async function run(): Promise<void> {
     const pathContext = core.getInput(Inputs.PATH_CONTEXT, { required: false });
     const logLevel = core.getInput(Inputs.LOG_LEVEL, { required: false });
     const envVars = core.getInput(Inputs.ENV_VARS, { required: false });
+    const includeGit = core.getInput(Inputs.INCLUDE_GIT, { required: false });
```
Comment (divyansh42) [inline src/s2iExec.ts:24]:
```suggestion
    const includeGit = core.getInput(Inputs.INCLUDE_GIT, { required: false }) | "false";
```

File: src/s2iExec.ts [hunk 2]
```diff
@@ -55,9 +56,14 @@ export async function run(): Promise<void> {
     }
 
     const buildCmd = [
-        "build", pathContext, builderImage, `${image}:${tagsList[0]}`, "--loglevel", logLevel,
+        "build", pathContext, builderImage, `${image}:${tagsList[0]}`, "--loglevel", logLevel
     ];
 
+    if (includeGit && includeGit != 'false') {
```
Comment (divyansh42) [inline src/s2iExec.ts:62]:
```suggestion
    if (includeGit === "true") {
```

File: src/generated/inputs-outputs.ts [hunk 3]
```diff
@@ -31,11 +31,17 @@ export enum Inputs {
      */
     PATH_CONTEXT = "path_context",
     /**
-     * The tags of the image to build. For multiple tags, seperate by a space. For example, "latest v1".
```
Comment (divyansh42) [inline src/generated/inputs-outputs.ts:34]: Please do not edit this manually.

**Review by divyansh42 — CHANGES_REQUESTED** [review]
Few nits that you may want to address.

**Comment by eformat** [comment]
> Few nits that you may want to address.

nice one thanks @divyansh42  .. done.

File: src/s2iExec.ts (hunk 1 above)
Comment (divyansh42) [reply src/s2iExec.ts:24]: It is not resolved yet. Could you please check once?

**Review by divyansh42 — CHANGES_REQUESTED** [review]

**Comment by divyansh42** [comment]
Also, can you add the link to the workflow where you have tested these changes?

**Comment by eformat** [comment]
ah soz, fixed initialization.
i have been testing in this workflow here

https://github.com/eformat/pet-battle-api/blob/master/.github/workflows/build_container.yaml#L40

**Review by divyansh42 — APPROVED** [review]
Thanks!

**Comment by divyansh42** [comment]
I've created a release for this feature. You can use this in `v2/v2.2`

**Pull request merged**